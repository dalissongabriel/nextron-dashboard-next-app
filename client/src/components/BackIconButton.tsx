import { ArrowBackOutlined } from "@mui/icons-material";
import { IconButton } from "@mui/material";
import Link from "next/link";

interface Props {
  href: string;
}

export default function BackIconButton({ href }: Props) {
  return (
    <IconButton color="secondary" LinkComponent={Link} href={href}>
      <ArrowBackOutlined />
    </IconButton>
  );
}
