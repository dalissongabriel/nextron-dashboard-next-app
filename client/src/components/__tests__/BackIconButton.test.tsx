import BackIconButton from "@components/BackIconButton";
import { render } from "@testing-library/react";

describe("BackIconButton", () => {
  it("should render back button correctly", () => {
    const { asFragment } = render(<BackIconButton href="#" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
