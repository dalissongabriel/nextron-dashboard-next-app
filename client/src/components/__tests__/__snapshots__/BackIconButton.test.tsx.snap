// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIconButton should render back button correctly 1`] = `
<DocumentFragment>
  <a
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-1gws2xf-MuiButtonBase-root-MuiIconButton-root"
    href="#"
    tabindex="0"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowBackOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </a>
</DocumentFragment>
`;
